be/src/vec/functions/function_regexp.cpp

```diff
@@ -50,6 +52,152 @@
+
+    size_t get_number_of_arguments() const override { return 2; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        auto int64_type = std::make_shared<DataTypeInt64>();
+        return make_nullable(std::static_pointer_cast<const IDataType>(int64_type));
+    }
+
+    Status open(FunctionContext* context, FunctionContext::FunctionStateScope scope) override {
+        if (scope == FunctionContext::THREAD_LOCAL && context->is_col_constant(1)) {
+            const auto pattern_col = context->get_constant_col(1)->column_ptr;
+            const auto& pattern = pattern_col->get_data_at(0);
+            if (pattern.size == 0) {
+                return Status::OK();
+            }
+            std::string error_str;
+            std::unique_ptr<re2::RE2> scoped_re;
+            bool st = StringFunctions::compile_regex(pattern, &error_str, StringRef(), StringRef(),
+                                                     scoped_re);
+            if (!st) {
+                context->set_error(error_str.c_str());
+                return Status::InvalidArgument(error_str);
+            }
+            std::shared_ptr<re2::RE2> re(scoped_re.release());
+            context->set_function_state(scope, std::static_pointer_cast<void>(re));
+        }
+        return Status::OK();
+    }
+
+    Status close(FunctionContext* context, FunctionContext::FunctionStateScope scope) override {
+        if (scope == FunctionContext::THREAD_LOCAL) {
+            auto ptr = context->get_function_state(scope);
+            if (ptr) {
+                delete reinterpret_cast<re2::RE2*>(ptr);
+                context->set_function_state(scope, nullptr);
+            }
+        }
+        return Status::OK();
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        uint32_t result, size_t input_rows_count) const override {
+        auto result_null_map = ColumnUInt8::create(input_rows_count, 0);
+        auto result_data_column = ColumnInt64::create();
+        auto& result_data = result_data_column->get_data();
+        result_data.resize(input_rows_count);
+        bool col_const[2];
+        ColumnPtr argument_columns[2];
+        for (int i = 0; i < 2; ++i) {
+            col_const[i] = is_column_const(*block.get_by_position(arguments[i]).column);
+        }
+        argument_columns[0] = col_const[0] ? static_cast<const ColumnConst&>(
+                                                     *block.get_by_position(arguments[0]).column)
+                                                     .convert_to_full_column()
+                                           : block.get_by_position(arguments[0]).column;
+
+        argument_columns[1] = col_const[1] ? static_cast<const ColumnConst&>(
+                                                     *block.get_by_position(arguments[1]).column)
+                                                     .convert_to_full_column()
```

Comment:
if argument_columns[0] and [1] are all need convert_to_full column,
why not 
for loop {
argument_columns[i] = block.get_by_position(arguments[0]).column).convert_to_full_column()
}